redash/models/__init__.py

```diff
@@ -315,6 +315,11 @@ def store_result(cls, org, data_source, query_hash, query, data, run_time, retri
                            retrieved_at=retrieved_at,
                            data=data)
         db.session.add(query_result)
+        db.session.commit()  # Make sure the query result is committed and assigned an ID.
+
+        if query_result.id is None:
+            raise Exception("inserted query result id is null for query_hash:" + query_hash)
```

Comment:
Wouldn't it be better if we were to be more specific about the type of Exception that is raised here? 
Also, are we sure it is handled in the calling function? what is the desired behaviour in this case?